framework/src/geomsearch/PenetrationThread.C

```diff
@@ -427,8 +427,12 @@ PenetrationThread::operator()(const NodeIdRange & range)
           } while (i < p_info.size() && best < p_info.size());
           if (best < p_info.size())
```

Comment:
```suggestion
          if (best < p_info.size() && CIResult != NEITHER_WINS)
```

Reply:
This would avoid needing to do that comparison again, and might make the code a little clearer since that checking for being within `_tangential_distance` would be done in one place in `competeInteractions()`.

You would also need to put something like `CompeteInteractionResult CIResult = NEITHER_WINS;` before this loop, around line 409, and then don't declare `CIResult` on line 412 anymore.

Reply:
I agree with this fix, but the results showed some problems. I think this should be explored in a different PR.
![image](https://github.com/user-attachments/assets/be75e189-ea97-420c-9e4c-f1677e9b0e87)